### PR TITLE
Multiple llvm versions 6+

### PR DESCRIPTION
**Pull request by andriish**
Support of standalone LLVM 6+.
De-facto that is 4+, as there is no difference between versions 4/5/6.

### CODE REVIEW

File: source/binder.cpp [hunk 1]
```diff
@@ -13,6 +13,8 @@
 #include <binder.hpp>
 
 // Declares clang::SyntaxOnlyAction.
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/ASTContext.h"
```
Comment (lyskov) [inline source/binder.cpp:17]: Could you please double check if these includes should not be need here? Or is it due to high version of LLVM  changes?

File: source/binder.hpp [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #define _INCLUDED_binder_hpp_
 
 
+#include "llvm/Config/llvm-config.h"
```
Comment (lyskov) [inline source/binder.hpp:18]: Optional: again, is this include really needed or is this result of removed LLVM support?

File: source/CMakeLists.txt [hunk 3]
```diff
@@ -33,6 +33,48 @@ add_clang_executable(binder
...
+                           LLVM-${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}
+                           PATHS  ${LLVM_LIBRARY_DIR}  ${LLVMLIBDIR}
+                           )
+add_definitions(${LLVM_DEFINITIONS})
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${LLVM_COMPILE_FLAGS}")
+list(REMOVE_ITEM CMAKE_CXX_FLAGS "-fno-exceptions")
+string (REPLACE "-fno-exceptions" ""     CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+message(STATUS "binder: LLVMSO=${LLVMSO}")
+message(STATUS "binder: CMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}")
+if ( (LLVM_VERSION_MAJOR GREATER 3)  AND (LLVM_VERSION_MAJOR LESS 8) )
```
[the remark this answers is not in the document]
Comment (andriish) [reply source/CMakeLists.txt:49]: There should be something in the with "greater" . But as 4 and 5 are technically the same as 6, 3 is therem

File: source/binder.cpp (hunk 1 above)
Comment (lyskov) [reply source/binder.cpp:17]: 👍

File: source/CMakeLists.txt [hunk 4]
```diff
@@ -33,6 +33,48 @@ add_clang_executable(binder
...
+  )
+endif()
+if ( (LLVM_VERSION_MAJOR EQUAL 10) OR (LLVM_VERSION_MAJOR GREATER 10) )
+target_link_libraries(binder
+  PRIVATE
+  ${lib_llvm_so_path}
+  clang
+  clang-cpp
+  )
+endif()
```
Comment (lyskov) [inline source/CMakeLists.txt:76]: please fix indentation in this block

File: source/CMakeLists.txt [hunk 5]
```diff
@@ -42,6 +84,7 @@ target_link_libraries(binder
   clangAST
   clangFrontend
   )
+endif()
```
Comment (lyskov) [inline source/CMakeLists.txt:87]: Please fix indentation in line range 78-86. Thanks,

File: source/binder.hpp (hunk 2 above)
Comment (lyskov) [reply source/binder.hpp:18]: Ah, i see it now, - the lack indentation confused me. Please see two comments regarding it.

File: source/CMakeLists.txt (hunk 5 above)
Comment (andriish) [reply source/CMakeLists.txt:87]: Done for the whole file.

Comment (lyskov) [reply source/CMakeLists.txt:87]: 👍

**Pull request merged**